app/models/agents/http_status_agent.rb

```diff
@@ -82,7 +82,8 @@ def check_this_url(url, local_headers)
 
       # Deal with failures
       if measured_result.result
-        payload.merge!({ 'response_received' => true, 'status' => current_status })
+        final_url = boolify(interpolated['disable_redirect_follow']) ? url : measured_result.result.to_hash[:url]
```

Comment:
@strugee Just looking at the code... perhaps ```measured_result.url``` would work here?  

The ```measured_result``` is a wrapper around the result, with a ```method_missing``` property to pass any method calls back to the original result.  (https://github.com/cantino/huginn/blob/29007c9019e35220964b16fb578c44c6effd0763/lib/time_tracker.rb#L15)

The other difference is that the tests have fake objects that may not have ```to_hash``` defined.  If ```to_hash``` is a method on the object in production, you could append ```to_hash``` to the fake and bingo it should be fine.

Sorry, this is a point where faking/mocking can be a pain.  And I say this as the one who wrote the original tests.  :smile: I thought the tests would be easier, since faking a bunch of different types of responses could be difficult or add more dependencies to Huginn.  But now the objects getting faked are more and more complicated.  This "given and take" decision is now firmly in "take" mode. My apologies on that. :sob: